marklogic-data-hub/src/main/resources/ml-modules/root/com.marklogic.smart-mastering/impl/sm-es-impl.xqy

```diff
@@ -97,6 +97,13 @@ declare function es-impl:get-entity-def($target-entity as item()?) as object-nod
       map:get($_cached-entities, $target-entity)
     else
       let $entity-def := fn:head(es-impl:get-entity-descriptors()/object-node()[(entityIRI,entityTitle) = $target-entity])
+      (: try a second time with title, if not with IRI found since external entities give different IRIs :)
```

Comment:
How easily can this be extracted into a separate function and tested? It seems like worthwhile logic to test, and often the comment that's here is better served as an assertion in a unit test.

Reply:
Added some basic assertions in a unit test